src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -184,7 +184,7 @@ public override Iterator<TSource> Clone()
 
             public override void Dispose()
             {
-                if (_enumerator is IDisposable) ((IDisposable)_enumerator).Dispose();
+                ((IDisposable)_enumerator).Dispose();
```

Comment:
I added a check for null before disposing the enumerator, but not casting to `IDisposable` may cause [backwards compatibility issues](http://stackoverflow.com/a/7051878/4077294) if it has an explicit interface implementation.